columnphysics/icepack_therm_vertical.F90

```diff
@@ -2589,7 +2601,8 @@ subroutine icepack_step_therm1(dt,                        &
          meltbn (n) = c0
          congeln(n) = c0
          snoicen(n) = c0
-         dsnown (n) = c0
+         l_dsnown   = c0
+         if (present(dsnown)) l_dsnown = dsnown (n)
```

Comment:
If the old implementation, "dsnown(n)" was always set to zero before being passed into thermo_vertical.  Now it is being initialized to the incoming value of "dsnown(n)".  Is this correct?  Why doesn't this change answers?

Reply:
This confused me as well, but dsnow and dsnown are both intent(inout). I guess the initialization to zero is automatically happening in the driver and we were never using it in the driver. I will fix this though.

Reply:
I am thinking that dsnown should be intent out only.

Reply:
Also, dsnown (called dsnow) is initialized to zero in thermo_vertical. Again, it should really only be intent(out).

Reply:
@dabail10, Please make updates ASAP if possible, would like to get this merged and into weekend testing as first steps to release.  Thanks.